### DISCUSSION TITLE
What to cache for CI using the PnP linker?

### DISCUSSION THREAD

**Discussion by jmariller** [yarnpkg/berry#4309] [Q&A] [answered]
The question was already asked for the node_modules linker [here](https://github.com/yarnpkg/berry/discussions/2621), but I am wondering what would be the proper setup when using the Plug'n'Play linker?

Right now I have one job to compile my assets, and another job to run tests with Jest. I have committed my `.yarn/.cache` folder, and tried to cache the full `.yarn` folder. Now when running the tests job I have to first run anyhow `yarn install`, otherwise jest cannot be found. Any ideas/suggestions?

**Comment by RDIL** [comment, marked as answer]
Given that you are using pnp and zero installs, you would also need to commit yarn.lock, .pnp.cjs, and .pnp.loader.mjs (I think that is what it’s called)
